### ISSUE TITLE
Requests fail when editing file with path that includes spaces

### ISSUE THREAD

**Issue by brdoney** [scalameta/metals#3794]
### Describe the bug

Whenever I edit a file that has a path which includes spaces, for example `~/Library/Mobile Documents/com~apple~CloudDocs/Documents/testing.sc` (note the space between `Mobile` and `Documents`), requests for folding ranges, code lenses, completion, and more seem to fail, even though the features are still provided as I'm actually editing the code.

I looked at the error log in VSCode and found that for nearly every request - `textDocument/foldingRange`, `textDocument/documentHighlight`, and `textDocument/codeLens`, to name a few - I get the same `URISyntaxException` at the root of the stack trace triggered by the failing request:

<details>
	<summary>Stack trace for when the textDocument/codeLens requests fail</summary>

```rawtext
java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: Illegal character in path at index 42: file:///Users/brdoney/Library/Mobile Documents/com~apple~CloudDocs/Documents/testing.sc
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$future$1(CancelTokens.scala:40)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$future$1$adapted(CancelTokens.scala:38)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

Caused by: java.lang.IllegalArgumentException: Illegal character in path at index 42: file:///Users/brdoney/Library/Mobile Documents/com~apple~CloudDocs/Documents/testing.sc
	at java.base/java.net.URI.create(URI.java:883)
	at scala.meta.internal.mtags.MtagsEnrichments$XtensionStringMtags.toAbsolutePath(MtagsEnrichments.scala:103)
	at scala.meta.internal.mtags.MtagsEnrichments$XtensionStringMtags.toAbsolutePath(MtagsEnrichments.scala:99)
	at scala.meta.internal.parsing.Trees.tokenized(Trees.scala:112)
	at scala.meta.internal.parsing.TokenEditDistance$.apply(TokenEditDistance.scala:383)
	at scala.meta.internal.metals.Buffers.tokenEditDistance(Buffers.scala:30)
	at scala.meta.internal.metals.codelenses.SuperMethodCodeLens.codeLenses(SuperMethodCodeLens.scala:38)
	at scala.meta.internal.metals.CodeLensProvider.$anonfun$findLenses$3(CodeLensProvider.scala:26)
	at scala.collection.immutable.List.flatMap(List.scala:366)
	at scala.meta.internal.metals.CodeLensProvider.$anonfun$findLenses$2(CodeLensProvider.scala:26)
	at scala.Option.map(Option.scala:230)
	at scala.meta.internal.metals.CodeLensProvider.findLenses(CodeLensProvider.scala:24)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$codeLens$2(MetalsLanguageServer.scala:1628)
	at scala.meta.internal.metals.TimerProvider.timedThunk(TimerProvider.scala:25)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$codeLens$1(MetalsLanguageServer.scala:1626)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$apply$2(CancelTokens.scala:26)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	... 4 more

Caused by: java.net.URISyntaxException: Illegal character in path at index 42: file:///Users/brdoney/Library/Mobile Documents/com~apple~CloudDocs/Documents/testing.sc
	at java.base/java.net.URI$Parser.fail(URI.java:2915)
	at java.base/java.net.URI$Parser.checkChars(URI.java:3086)
	at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3168)
	at java.base/java.net.URI$Parser.parse(URI.java:3116)
	at java.base/java.net.URI.<init>(URI.java:600)
	at java.base/java.net.URI.create(URI.java:881)
	... 25 more
```
</details>

TLDR: there's a `java.net.URISyntaxException` because of a space character at index 42 of the file path.

### Expected behavior

Metals works seamlessly for anything with a space in it, probably by encoding the URL before using it.

### Operating system

macOS

### Editor/Extension

VS Code

### Version of Metals

v0.11.2
